### PR TITLE
Enable build with bindings

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

**Comment by m-kuhn** [comment]
@domi4484 FYI something is needed here too

```
sip-build: D:\a\kadas-albireo2\kadas-albireo2\build\python\kadascore\kadascore.sip: line 10: column 9: 'core/core.sip' could not be found
```

File: vcpkg/ports/vcpkg-python-scripts/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "vcpkg-python-scripts",
```
Comment (m-kuhn) [inline vcpkg/ports/vcpkg-python-scripts/vcpkg.json:2]: Note to self: remove this port before merge

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,9 +18,18 @@ if (WITH_QTWEBKIT)
...
 
+set(WITH_BINDINGS ON CACHE BOOL "Build with python bindings")
+if (WITH_BINDINGS)
+  list(APPEND VCPKG_MANIFEST_FEATURES "bindings")
+endif()
+
 include(VcpkgToolchain)
 
+# The default installation paths
+include(GNUInstallDirs)
```
Comment (m-kuhn) [inline CMakeLists.txt:29]: TODO:

1. move after project call
2. add `CXX` lang to project

**Comment by m-kuhn** [comment]
@3nids I'd like to get the green light for this, could you do a review?
I am also available for a pair review if you think it makes sense.

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -50,17 +55,33 @@ jobs:
           path: |
             C:/src/**/*.log
 
+      # TODO: only on release or workflow dispatch
+      - name: 📦 Create SDK
+        if: github.event_name == 'pull_request'
```
Comment (3nids) [inline .github/workflows/windows.yml:60]:
```suggestion
        if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
```

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -50,17 +55,33 @@ jobs:
...
+        run: |
+          ./build/_deps/vcpkg-src/vcpkg.exe export --zip --output-dir=./sdk --x-install-root=build/vcpkg_installed
+        
+      - name: 📤 Upload sdk
+        if: github.event_name == 'pull_request'
```
Comment (3nids) [inline .github/workflows/windows.yml:65]:
```suggestion
        if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
```

File: .github/workflows/windows.yml [hunk 5]
```diff
@@ -50,17 +55,33 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: kadas-albireo2-sdk-x64-windows
+          path: |
+            sdk/vcpkg-export-*.zip
+      # TODO: end
```
Comment (3nids) [inline .github/workflows/windows.yml:71]: ?

File: cmake/FindQGIS.cmake [hunk 6]
```diff
@@ -86,6 +89,7 @@ if(Core IN_LIST QGIS_FIND_COMPONENTS)
 endif()
 if(Analysis IN_LIST QGIS_FIND_COMPONENTS)
   _find_qgis_library(analysis Analysis)
+  target_link_libraries(QGIS::Analysis INTERFACE QGIS::Core)
```
Comment (3nids) [inline cmake/FindQGIS.cmake:92]: there is no target_link_libraries for QGIS::Gui, is this expected?

File: cmake/VcpkgInstallDeps.cmake [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+if(NOT EXISTS "${PROJ_DATA_PATH}/proj.db")
+    message(FATAL_ERROR "proj.db not found at ${PROJ_DATA_PATH}/proj.db")
+endif()
+
+copy_resource("${PROJ_DATA_PATH}" "proj")
+copy_resource("${VCPKG_BASE_DIR}/share/gdal" "gdal")
+copy_resource("${CMAKE_SOURCE_DIR}/packaging/files" "kadas")
+#install(DIRECTORY "${SHARE_DIR}/qgis/resources/" DESTINATION "${CMAKE_INSTALL_DATADIR}/qgis/resources")
+#install(DIRECTORY "${QGIS_SHARE_DIR}/svg/" DESTINATION "${CMAKE_INSTALL_DATADIR}/qgis/svg")
```
Comment (3nids) [inline cmake/VcpkgInstallDeps.cmake:110]: no need for the SVGs?

File: vcpkg/ports/qgis/bindings-install.patch [hunk 8]
```diff
@@ -0,0 +1,59 @@
+diff --git a/cmake/SIPMacros.cmake b/cmake/SIPMacros.cmake
```
Comment (3nids) [inline vcpkg/ports/qgis/bindings-install.patch:1]: I haven't look much here, but is this something that can be put upstream or we have to maintain it?

**Review by 3nids — APPROVED** [review]

File: .github/workflows/windows.yml (hunk 3 above)
Comment (m-kuhn) [reply .github/workflows/windows.yml:60]:
```suggestion
        if: github.event_name == 'workflow_dispatch'
```

File: .github/workflows/windows.yml (hunk 4 above)
Comment (m-kuhn) [reply .github/workflows/windows.yml:65]:
```suggestion
        if: github.event_name == 'workflow_dispatch'
```

File: .github/workflows/windows.yml (hunk 5 above)
Comment (m-kuhn) [reply .github/workflows/windows.yml:71]:
```suggestion
```

File: cmake/VcpkgInstallDeps.cmake (hunk 7 above)
Comment (m-kuhn) [reply cmake/VcpkgInstallDeps.cmake:110]: Not sure ... to be clarified.

File: cmake/FindQGIS.cmake (hunk 6 above)
Comment (m-kuhn) [reply cmake/FindQGIS.cmake:92]: Well spotted

File: cmake/VcpkgInstallDeps.cmake (hunk 7 above)
Comment (m-kuhn) [reply cmake/VcpkgInstallDeps.cmake:110]: FTR: this adds many files which slow down the upload of the test artifact, that's why I disabled this.
I think we could also zip the files before uploading which would speed up things.

File: CMakeLists.txt [hunk 9]
```diff
@@ -88,6 +98,8 @@ IF(INSTALL_DEMO_DATA)
     install(DIRECTORY packaging/files/print_templates packaging/files/project_templates DESTINATION ${CMAKE_INSTALL_PREFIX}/share/kadas/)
 ENDIF()
 
+include(VcpkgInstallDeps)
+
 IF(UNIX)
     install(FILES com.sourcepole.kadas.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications/)
```
Comment (3nids) [inline CMakeLists.txt:104]:
```suggestion
    install(FILES com.kadas.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications/)
```
?

File: CMakeLists.txt [hunk 10]
```diff
@@ -88,6 +98,8 @@ IF(INSTALL_DEMO_DATA)
...
     install(FILES com.sourcepole.kadas.appdata.xml DESTINATION ${CMAKE_INSTALL_PREFIX}/share/metainfo/)
```
Comment (3nids) [inline CMakeLists.txt:105]: idem

File: CMakeLists.txt (hunk 9 above)
Comment (m-kuhn) [reply CMakeLists.txt:104]: as this is not related to bindings, I'd rather keep that separate for now?

**Comment by m-kuhn** [comment]
@3nids the latest commit should fix the build issues for you (clang 16)

**Pull request merged**